docs/sbft-formal-model/proof.dfy

```diff
@@ -347,14 +346,44 @@ module Proof {
     }
     assert f + 1 <= |commonSenders|;
 
-    var honestReplicas := set sender | 0 <= sender < n && IsHonestReplica(c, sender);
-    var commonHonest := commonSenders * honestReplicas;
-    assert 1 <= |commonHonest|;
+    var commonHonest := commonSenders * HonestReplicas(c);
+    CountHonestReplicas(c);
+    if(|commonHonest| == 0) {
+      Library.SubsetCardinality(commonSenders + HonestReplicas(c), getAllReplicas(c));
+      assert false; //Proof by contradiction
+    }
 
     var result :| result in commonHonest;
     common := result;
   }
 
+  function HonestReplicas(c: Constants) : (honest:set<HostId>) 
+    requires c.WF()
+  {
+    set sender | 0 <= sender < c.clusterConfig.N() && IsHonestReplica(c, sender)
+  }
+
+  lemma CountHonestReplicas(c: Constants)
+    requires c.WF()
+    ensures |HonestReplicas(c)| == c.clusterConfig.AgreementQuorum()
```

Comment:
shouldn't this rather be `|HonestReplicas(c)| >= c.clusterConfig.AgreementQuorum()` ?

Reply:
Currently we set the faulty replicas to be the maximum safety preserving value and the honest are exactly = 2F+1, but it is more accurate to make this >=, because we might want to run the verifier with less than F faulty also. I will change this accordingly.

Reply:
Thanks, sounds good! I hope it doesn't break the existing proofs...

Reply:
I checked locally, the proof passes correctly. This reminds me that we need to add the check for the proof (and syntactic check for all .dfy files) in the CI soon :)